src/engine/structures.rs

```diff
@@ -0,0 +1,156 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use std::collections::HashMap;
+use std::slice::IterMut;
+
+use uuid::Uuid;
+
+use super::engine::{HasName, HasUuid};
```

Comment:
It is quite possible, but we should defer that to a later PR and when we've looked at a few more cases.